polaris-service/src/main/java/org/apache/polaris/service/config/PolarisApplicationConfig.java

```diff
@@ -63,69 +88,206 @@ public class PolarisApplicationConfig extends Configuration {
+      protected void configure() {
+        bindFactory(SupplierFactory.create(serviceLocator, config::getStorageIntegrationProvider))
+            .to(PolarisStorageIntegrationProvider.class)
+            .ranked(OVERRIDE_BINDING_RANK);
+        bindFactory(SupplierFactory.create(serviceLocator, config::getMetaStoreManagerFactory))
+            .to(MetaStoreManagerFactory.class)
+            .ranked(OVERRIDE_BINDING_RANK);
+        bindFactory(SupplierFactory.create(serviceLocator, config::createConfigurationStore))
+            .to(PolarisConfigurationStore.class)
+            .ranked(OVERRIDE_BINDING_RANK);
+        bindFactory(SupplierFactory.create(serviceLocator, config::getFileIOFactory))
+            .to(FileIOFactory.class)
+            .ranked(OVERRIDE_BINDING_RANK);
+        bindFactory(SupplierFactory.create(serviceLocator, config::getPolarisAuthenticator))
+            .to(Authenticator.class)
+            .ranked(OVERRIDE_BINDING_RANK);
+        bindFactory(SupplierFactory.create(serviceLocator, config::getTokenBrokerFactory))
+            .to(TokenBrokerFactory.class)
+            .ranked(OVERRIDE_BINDING_RANK);
+        bindFactory(SupplierFactory.create(serviceLocator, config::getOauth2Service))
+            .to(IcebergRestOAuth2ApiService.class)
+            .ranked(OVERRIDE_BINDING_RANK);
+        bindFactory(SupplierFactory.create(serviceLocator, config::getCallContextResolver))
+            .to(CallContextResolver.class)
+            .ranked(OVERRIDE_BINDING_RANK);
+        bindFactory(SupplierFactory.create(serviceLocator, config::getRealmContextResolver))
+            .to(RealmContextResolver.class)
+            .ranked(OVERRIDE_BINDING_RANK);
+        bindFactory(SupplierFactory.create(serviceLocator, config::getRateLimiter))
+            .to(RateLimiter.class)
+            .ranked(OVERRIDE_BINDING_RANK);
+      }
+    };
+  }
+
+  /**
+   * Factory implementation that uses the provided supplier method to retrieve the instance and then
+   * uses the {@link #serviceLocator} to inject dependencies into the instance. This is necessary
+   * since the DI framework doesn't automatically inject dependencies into the instances created.
+   *
+   * @param <T>
+   */
+  private static final class SupplierFactory<T> implements Factory<T> {
+    private final ServiceLocator serviceLocator;
+    private final Supplier<T> supplier;
+
+    private static <T> SupplierFactory<T> create(
+        ServiceLocator serviceLocator, Supplier<T> supplier) {
+      return new SupplierFactory<>(serviceLocator, supplier);
+    }
+
+    private SupplierFactory(ServiceLocator serviceLocator, Supplier<T> supplier) {
+      this.serviceLocator = serviceLocator;
+      this.supplier = supplier;
+    }
+
+    @Override
+    public T provide() {
+      T obj = supplier.get();
+      serviceLocator.inject(obj);
+      return obj;
+    }
+
+    @Override
+    public void dispose(T instance) {}
+  }
+
+  public <T> T findService(Class<T> serviceClass) {
```

Comment:
This seems to be a CDI concern rather than a configuration concern. I mean that it is perfectly fine to drive CDI with config settings, but `findService` does not appear to fit the DI paradigm :thinking:

Reply:
It doesn't, but the servlet filters don't use DI, so the PolarisApplication has to find them somehow